### PR TITLE
datovka, libisds: init at 4.12.0 and 0.10.8; added maintainer cptMikky

### PR DESCRIPTION
**Pull request by jficz**
###### Motivation for this change

Add new package `Datovka` (https://www.datovka.cz) useful to anyone who has to communicate with Czech government via Databoxes.

Add new pacakge `libisds` (http://xpisar.wz.cz/libisds/) which `Datovka` depends on.

Add new maintainer, @cptMikky .
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by cptMikky** [comment]
Just a preview, Datovka doesn't build.

```
g++ -c -g -O0 -std=c++11 -Wall -Wextra -pedantic -Wdate-time -Wformat -Werror=format-security -O2 -Wall -W -D_REENTRANT -fPIC -DDEBUG=1 -DVERSION=\"4.12.0\" -DDATADIR=\"/nix/store/7gc3dryzi5xw41b42gvi5sp9d8285ndz-datovka-4.12.0/share\" -DPKGDATADIR=\"\" -DLOCALE_INST_DIR="\"/nix/store/7gc3dryzi5xw41b42gvi5sp9d8285ndz-datovka-4.12.0/share/datovka/localisations\"" -DTEXT_FILES_INST_DIR="\"/nix/store/7gc3dryzi5xw41b42gvi5sp9d8285ndz-datovka-4.12.0/share/doc/datovka\"" -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_SQL_LIB -DQT_CORE_LIB -I. -I/usr/include/libxml2 -isystem /nix/store/ywqn919y4lnmfy2dpk5zx1bd6pkkv9dg-qtsvg-5.12.0-dev/include/QtSvg -isystem /nix/store/9ghnbhx3qwdawci3p24yc4axxblwjhbw-qtbase-5.12.0-dev/include/QtPrintSupport -isystem /nix/store/9ghnbhx3qwdawci3p24yc4axxblwjhbw-qtbase-5.12.0-dev/include/QtWidgets -isystem /nix/store/9ghnbhx3qwdawci3p24yc4axxblwjhbw-qtbase-5.12.0-dev/include/QtGui -isystem /nix/store/9ghnbhx3qwdawci3p24yc4axxblwjhbw-qtbase-5.12.0-dev/include/QtNetwork -isystem /nix/store/9ghnbhx3qwdawci3p24yc4axxblwjhbw-qtbase-5.12.0-dev/include/QtSql -isystem /nix/store/9ghnbhx3qwdawci3p24yc4axxblwjhbw-qtbase-5.12.0-dev/include/QtCore -Igen_moc -I/nix/store/vl9ma114n5xh3q7p9vjhl1p7vzg3vp1r-mesa-noglu-18.3.4-dev/include -Igen_ui -I/nix/store/9ghnbhx3qwdawci3p24yc4axxblwjhbw-qtbase-5.12.0-dev/mkspecs/linux-g++ -o gen_objects/src/cli/cli_parser.o src/cli/cli_parser.cpp
In file included from src/about.cpp:32:0:
/nix/store/cvx6vfnl6wc89l5cigvz3rlvh85rbb90-libisds-0.10.8/include/isds.h:10:10: fatal error: libxml/tree.h: No such file or directory
 #include <libxml/tree.h>    /* For xmlDoc and xmlNodePtr */
          ^~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:2225: gen_objects/src/about.o] Error 1
make: *** Waiting for unfinished jobs....
builder for '/nix/store/gkvm2g4hv9svvi901va7gvhins2r2p88-datovka-4.12.0.drv' failed with exit code 2

```

**Comment by cptMikky** [comment]
Updated, now builds and runs.

**Comment by cptMikky** [comment]
Should I squash the changes? Or maybe separate them to three commits (add maintainer, add libisds, add datovka)? Or should I just leave the PR as it is?

**Comment by ryantm** [comment]
Three commits would be good. Also, please format them to match https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md

File: pkgs/applications/networking/datovka/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, fetchurl, libxml2, libisds, qmake, qtbase, qtsvg }:
+
+stdenv.mkDerivation rec {
+  name = "datovka-${version}";
```
Comment (ryantm) [inline pkgs/applications/networking/datovka/default.nix]:
```suggestion
  pname = "datovka";
```

File: pkgs/applications/networking/datovka/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  version = "4.12.0";
+
+  src = fetchurl {
+    url = "https://secure.nic.cz/files/datove_schranky/${version}/${name}.tar.xz";
```
Comment (ryantm) [inline pkgs/applications/networking/datovka/default.nix]:
```suggestion
    url = "https://secure.nic.cz/files/datove_schranky/${version}/${pname}-${version}.tar.xz";
```

File: pkgs/development/libraries/libisds/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchurl, expat, gpgme, libgcrypt, libxml2 }:
+
+stdenv.mkDerivation rec {
+  name = "libisds-${version}";
```
Comment (ryantm) [inline pkgs/development/libraries/libisds/default.nix]:
```suggestion
  pname = "libisds";
```

File: pkgs/development/libraries/libisds/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  version = "0.10.8";
+
+
+
+  src = fetchurl {
+    url = "http://xpisar.wz.cz/libisds/dist/${name}.tar.xz";
```
Comment (ryantm) [inline pkgs/development/libraries/libisds/default.nix]:
```suggestion
    url = "http://xpisar.wz.cz/libisds/dist/${pname}-${version}.tar.xz";
```

File: pkgs/development/libraries/libisds/default.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    sha256 = "1av6i7acqrsl054mm6hbz20qcnbmxmzbax942c31576f00vwcqbq";
+  };
+
+  enableParallelBuilding = true;
+
+  buildInputs = [ expat gpgme libgcrypt ];
+  propagatedBuildInputs = [ libxml2 ];
```
Comment (ryantm) [inline pkgs/development/libraries/libisds/default.nix]: Have you tried just having this be a build input?

**Review by ryantm — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/libisds/default.nix (hunk 5 above)
Comment (ryantm) [reply pkgs/development/libraries/libisds/default.nix]: If that doesn't work, I expect the proper thing to do is use makeWrapper to make the library available where needed.

Comment (jtojnar) [reply pkgs/development/libraries/libisds/default.nix]: It should be fine. It is in the `Requires` section of pc file.

File: pkgs/applications/networking/datovka/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    sha256 = "1dcs6yk7dflk1dxxbn5hjgc008bpgiqfcgy0cj0fxnpdn0i7za57";
+  };
+
+  enableParallelBuilding = true;
+
+  buildInputs = [ libisds qmake qtbase qtsvg libxml2 ];
+  NIX_CFLAGS_COMPILE = [ "-I${libxml2.dev}/include/libxml2/" ];
```
Comment (jtojnar) [inline pkgs/applications/networking/datovka/default.nix]: Can you add a comment linking to https://gitlab.labs.nic.cz/datovka/datovka/issues/409?

**Comment by mmahut** [comment]
@cptMikky any update on this pull request?

**Comment by mmahut** [comment]
Closing due to inactivity, feel free to re-open this pull request.

**Pull request closed without merging**